Refactor "fix" / Improvments

MatiasPastori opened the pull request:
More refactor

NVG preset changes, NVG automatic switch in or out vehicle.

Added ShaderComponents to the presets, (now being interpreted by VEM too)

Clean up patchAssets

UI changes, animation, sounds, etc.

MatiasPastori committed:
Fixed the NVG state control

Created a preset specifically for NVG in Vehicles (Still needs work to be functional)

Added some configs and flags so its easier to communicate some things such as Time managment in VEM from DU.

Ended up not using the MapVEManager module, so this should be deleted at some point of the future.

MatiasPastori committed:
Now it works correctly with VEManager, fun-bots and other mods such as compass.

Implemented the automatic switch of the nvg when we get in or out of a vehicle.

Added the thermal vision to the vehicle nvg preset.

Added config options for vehicle lights.

Removed the patchAssets and the patching we were doing for meshes and other stuff that seems unnecesserary now that VEM already does that without issues.

TODO: Add the vehicle lights but make them toggleable and default them to be off.

MatiasPastori committed:
Updated dependencies.

Refactor and cleanup

Small changes in the NVG and Vehicle NVG presets.

Removed the PatchAssets script since we don't need it anymore and it was causing issues with VEM since its already patching most of the stuff.

Kept the patches script since we are still doing some things that VEM doesn't and maybe it shouldn't at all, for example, the bullet, impacts, explossions emitters, flashlights and such.

Added some UI stuff, a cute animation for the situation where the NVG is depleted so it bounces the nvg goggle icon and some red color.
Also changed some audio files for the NVG.

TODO: I'd like to be able to "pulse" or make some kind of glitch preset for a couple seconds if the player tries to turn on the NVG while still on cooldown (Tried using Pulse from VEM but it wasn't working properly)

Config file cleanup, the time settings at least for now. Enabling time as soon as the server can will cause the game to crash as soon as a player gets in, still don't know the cause of this.

MatiasPastori committed:
Fixed some preset stuff so the transitions look clean and smooth :)

Logger refactor due to the fact that there are already a lot of mods using or implementing their one Loggers and they are all called the same, way, so with the a more precise name for the config variables and Logger it shoudln't cause any issues.

*enlighten seems to cause visual issues or glitches when transitioning from one preset to another, so its disabled for all of them for now*

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_001/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
unnecessary?

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_001/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
 		"Enable":"true",
 		"SunRotationX":"0.0",
-		"SunRotationY":"180.0",
+		"SunRotationY": "180.0",
```
same here?

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_003/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
same here

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_003/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
 		"Enable":"true",
 		"SunRotationX":"0.0",
-		"SunRotationY":"180.0",
+		"SunRotationY": "180.0",
```
same here

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_007/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
same here

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_007/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
        	"Enable":"true",
        	"SunRotationX":"0.0",
-		"SunRotationY":"180.0",
+		"SunRotationY": "180.0",
```
what happened to all of those? :D

MatiasPastori replied:
Already cleaned that up in the last commit

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_011/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
        "Enable":"true",
        "SunRotationX":"0.0",
-       "SunRotationY":"180.0",
+       "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_012/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_012/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
       "Enable":"true",
       "SunRotationX":"0.0",
-      "SunRotationY":"180.0",
+      "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_013/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_017/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_017/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_018/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_018/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_Subway/NVG.lua:
```diff
@@ -90,7 +90,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

IllustrisJack commented on ext/Client/Presets/Vanilla/MP_Subway/Night.lua:
```diff
@@ -77,7 +77,7 @@ return [[
         "Enable":"true",
         "SunRotationX":"0.0",
-        "SunRotationY":"180.0",
+        "SunRotationY": "180.0",
```
-

MatiasPastori committed:
PR: Review

MatiasPastori commented:
@IllustrisJack You were right, I messed up with that extra space ( just me editing multiple files at once)

MatiasPastori committed:
Using the new fix in VEM that allows it to actually use the X axis coordinates on the presets :)

Also added an example of a morning preset for MP_007
We could create a 5 preset pack and avoid using the default ones if needed. Depends on what is the objective of whoever using this.
